frame/system/src/lib.rs

```diff
@@ -1963,8 +1965,11 @@ pub(crate) mod tests {
 			System::deposit_event(32u16);
 			System::note_finished_initialize();
 			System::deposit_event(42u16);
-			System::note_applied_extrinsic(&Ok(()), 0, Default::default());
-			System::note_applied_extrinsic(&Err(DispatchError::BadOrigin), 0, Default::default());
+			System::note_applied_extrinsic(&Ok(().into()), Default::default());
+			System::note_applied_extrinsic(
+				&Err(DispatchError::BadOrigin.into()),
+				Default::default()
+			);
```

Comment:
let's add a test that checks the weight?

Reply:
With the current testing infrastructure in this module this is not possible because all Events are converted to `u16` in the test `Trait` configuration. This looses any information that is stored in the Event apart from its index.

Reply:
I would personally see it as a very valuable follow up to consider making these test-able. I think working on unit tests or a bit underrated, they're super critical and important to review as well.

Reply:
So you want to me to refactor that whole test suite? 😱

Reply:
Only one or two tests work with events, should be rather easy. I can help if you want.

Reply:
Look for other usages of `decl_event! {}` in `mod tests`

Reply:
Done. Converted to the proper event type and added a new test.